src/screens/authentication/UsernameRequest/index.js

```diff
@@ -19,70 +42,66 @@ const SubmitIndicator = ({ submitting }) => (
     />
 )
 
-const Error = ({ error }) => (
-  error && (
-    <Body
-      secondary
-      style={styles.error}
-    >
-      {error}
-    </Body>
-  )
+const SubmitButton = ({ disabled, onPress, error }) => (
+  <PrimaryButton
+    onPress={onPress}
+    style={error ? { marginTop: verticalScale(20) } : { marginTop: verticalScale(40) }}
+    disabled={error ? true : disabled}
+  >
+    <Description white>Continuar</Description>
+  </PrimaryButton>
 )
 
-const SubmitButton = ({ disabled, onPress }) => {
-  if (disabled) {
-    return (
-      <GrayButton>
-        <Description white>Continuar</Description>
-      </GrayButton>
-    )
-  } else {
-    return (
-      <PrimaryButton
-        onPress={onPress}
-      >
-        <Description white>Continuar</Description>
-      </PrimaryButton>
-    )
-  }
-}
-
 const Icon = () => (
-  <>
-    <View style={styles.iconBackground} />
-    <SvgXml
-      width={ICON_WIDTH}
-      height={ICON_HEIGHT}
-      style={styles.icon}
-      xml={usernameIcon}
-    />
-  </>
+  <SvgXml
+    width={ICON_WIDTH}
+    height={ICON_HEIGHT}
+    style={styles.icon}
+    xml={usernameIcon}
+  />
+)
+
+const Suggestions = ({ error, suggestions, onSuggestionSelected }) => (
+  error && (
+    <View style={styles.suggestionContainer}>
+      {suggestions.map((suggestion, index) => {
+        if (index < 4) {
```

Comment:
This `if` is weird, it's purpose is to prevent the map from returning something. If I want to add more elements to `suggestions` I would have to check the code as to why my new element is not rendering. Please remove it and in any case force `suggestions` to always be a 4-length array